mathesar_ui/src/systems/table-view/TableView.svelte

```diff
@@ -1,39 +1,41 @@
 <script lang="ts">
-  import type { TableEntry } from '@mathesar/api/tables';
-  import type { Database, SchemaEntry } from '@mathesar/AppTypes';
   import { ImmutableMap } from '@mathesar/component-library';
   import { Sheet } from '@mathesar/components/sheet';
   import {
+    getTabularDataStoreFromContext,
     ID_ADD_NEW_COLUMN,
     ID_ROW_CONTROL_COLUMN,
-    setTabularDataStoreInContext,
   } from '@mathesar/stores/table-data';
-  import type { TabularData } from '@mathesar/stores/table-data/types';
-  import { writable } from 'svelte/store';
-  import ActionsPane from './actions-pane/ActionsPane.svelte';
   import Body from './Body.svelte';
   import Header from './header/Header.svelte';
   import StatusPane from './StatusPane.svelte';
   import TableInspector from './table-inspector/TableInspector.svelte';
 
-  export let database: Database;
-  export let schema: SchemaEntry;
-  export let tabularData: TabularData;
-  export let table: TableEntry;
+  const tabularData = getTabularDataStoreFromContext();
```

Comment:
Is there an advantage to getting tabularData from context here, instead of a prop?

Since TableView is a system on it's own and tabularData is the primary data source for this system, it is more intuitive to use a prop instead unless there's a good reason not to.

Reply:
Before this PR, `TableView.svelte` was setting `tabularData` in context. After this PR, it's _getting_ `tabularData` from context. I made that change because I moved `ActionsPane` outside of `TableView` and `ActionsPane` already depended on `tabularData` being in context. Moving the location at which the context is set seemed easier than chasing down all the context dependencies within `ActionsPane`. I would be open to establishing a standard that _"the top-level component of a system does not depend on context"_, but it's not how I've been operating thus far.